packages/browser/test/client.test.ts

```diff
@@ -1,38 +1,134 @@
+import { Source } from '../src/config';
 import { ExperimentClient } from '../src/experimentClient';
-import { ExperimentUser } from '../src/types/user';
+import { ExperimentUser, ExperimentUserProvider } from '../src/types/user';
+import { Variant, Variants } from '../src/types/variant';
+
+const delay = (ms: number) => new Promise((res) => setTimeout(res, ms));
+
+class TestUserProvider implements ExperimentUserProvider {
+  getUser(): ExperimentUser {
+    return testUser;
+  }
+}
 
 const API_KEY = 'client-DvWljIjiiuqLbyjqdvBaLFfEBrAvGuA3';
 
 const testUser: ExperimentUser = { user_id: 'test_user' };
-const testClient: ExperimentClient = new ExperimentClient(API_KEY, {});
 
-const testTimeoutNoRetriesClient = new ExperimentClient(API_KEY, {
-  retryFetchOnFailure: false,
-  fetchTimeoutMillis: 1,
-});
+const fallbackVariant: Variant = { value: 'fallback', payload: 'payload' };
+const serverVariant: Variant = { value: 'on', payload: 'payload' };
+const initialVariant: Variant = { value: 'initial' };
+const offVariant: Variant = { value: 'off' };
 
-const testTimeoutRetrySuccessClient = new ExperimentClient(API_KEY, {
-  fetchTimeoutMillis: 1,
-});
+const key = 'sdk-ci-test';
+const initialKey = 'initial-key';
+
+const initialVariants: Variants = {
+  'initial-key': initialVariant,
+  'sdk-ci-test': offVariant,
+};
 
 beforeEach(() => {
   localStorage.clear();
 });
 
 test('ExperimentClient.fetch, success', async () => {
-  await testClient.fetch(testUser);
-  const variant = testClient.variant('sdk-ci-test');
+  const client = new ExperimentClient(API_KEY, {});
+  await client.fetch(testUser);
+  const variant = client.variant('sdk-ci-test');
   expect(variant).toEqual({ value: 'on', payload: 'payload' });
 });
 
 test('ExperimentClient.fetch, no retries, timeout failure', async () => {
-  await testTimeoutNoRetriesClient.fetch(testUser);
-  const variants = testTimeoutNoRetriesClient.all();
+  const client = new ExperimentClient(API_KEY, {
+    retryFetchOnFailure: false,
+    fetchTimeoutMillis: 1,
+  });
+  await client.fetch(testUser);
+  const variants = client.all();
   expect(variants).toEqual({});
 });
 
-test('ExperimentClient.fetch, no retries, timeout failure', async () => {
-  await testTimeoutRetrySuccessClient.fetch(testUser);
-  const variant = testClient.variant('sdk-ci-test');
+test('ExperimentClient.fetch, with retries, retry success', async () => {
+  const client = new ExperimentClient(API_KEY, {
+    fetchTimeoutMillis: 1,
+  });
+  await client.fetch(testUser);
```

Comment:
Does this fetch wait from the retry?

Reply:
No, the retries occur in the background.

Reply:
Could you assert that `variant('sdk-ci-test')` at this point returns the config fallback? I think that would help demonstrate the intention of the test.